src/System.IO.Pipelines/tests/StreamPipeReaderTests.cs

```diff
@@ -0,0 +1,438 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Pipelines.Tests
+{
+    public class StreamPipeReaderTests
```

Comment:
Do you have tests wrapping a stream that actually goes async and then returns some data?

Reply:
Yes, the ones that use Pipe https://github.com/dotnet/corefx/pull/36575/files#diff-65f2e511c602e4946378bdb12409969eR288.